zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -182,6 +192,13 @@ protected void addChangeRecord(ChangeRecord c) {
         }
     }
 
+    protected void decrementEphemeralCount(ChangeRecord record) {
+        if (ephemeralCount.containsKey(record.stat.getEphemeralOwner())) {
```

Comment:
What do you think about replacing this logic with `compute()` method of `ConcurrentHashMap`?
I'm not sure if we need to synchronize here (probably not), but the code would be cleaner.

Reply:
Good suggestion. Wasn't aware of `compute()`.